onnxruntime/core/providers/rocm/nn/conv.cc

```diff
@@ -352,9 +352,9 @@ Status Conv<T, NHWC>::ComputeInternal(OpKernelContext* context) const {
       const Tensor* B = context->Input<Tensor>(2);
       const auto& b_shape = B->Shape();
 
-      ConvBiasImpl(Stream(context), reinterpret_cast<HipT*>(s_.Y->MutableData<T>()),
+      ConvBiasImpl(Stream(context), reinterpret_cast<const HipT*>(s_.y_data),
                    reinterpret_cast<const HipT*>(B->Data<T>()),
-                   reinterpret_cast<HipT*>(s_.Y->MutableData<T>()), b_shape[0], s_.Y->Shape().Size());
+                   reinterpret_cast<HipT*>(s_.y_data), b_shape[0], s_.Y->Shape().Size());
```

Comment:
I am curious why the unit test of NhwcConv does not catch this bug. Could you double check the test and update if necessary?

Reply:
`s_.y_data` will use `an intermediate buffer` if post slicing is needed(asymmetric padding), and will use `s_.Y->MutableData<T>()` if post slicing is not needed(symmetric padding). The unit case doesn't test the combination of asymmetric padding + bias.